Sources/Extensions/SwiftStdlib/RangeReplaceableCollectionExtensions.swift

```diff
@@ -0,0 +1,50 @@
+//
+//  RangeReplaceableCollectionExtensions.swift
+//  SwifterSwift
+//
+//  Created by Luciano Almeida on 7/2/18.
+//  Copyright © 2018 SwifterSwift
+//
+
+// MARK: - Methods
+extension RangeReplaceableCollection {
+
+    /// SwifterSwift: Returns a new rotated collection by the given places.
+    ///
+    ///     [1, 2, 3, 4].rotated(by: 1) -> [4,1,2,3]
+    ///     [1, 2, 3, 4].rotated(by: 3) -> [2,3,4,1]
+    ///     [1, 2, 3, 4].rotated(by: -1) -> [2,3,4,1]
+    ///
+    /// - Parameter places: Number of places that the array be rotated. If the value is positive the end becomes the start, if it negative it's that start becom the end.
+    /// - Returns: The new rotated collection.
+    public func rotated(by places: Int) -> Self {
+        //Inspired by: https://ruby-doc.org/core-2.2.0/Array.html#method-i-rotate
+        var copy = self
+        return copy.rotate(by: places)
+    }
+
+    /// SwifterSwift: Rotate the collection by the given places.
+    ///
+    ///     [1, 2, 3, 4].rotate(by: 1) -> [4,1,2,3]
+    ///     [1, 2, 3, 4].rotate(by: 3) -> [2,3,4,1]
+    ///     [1, 2, 3, 4].rotated(by: -1) -> [2,3,4,1]
+    ///
+    /// - Parameter places: The number of places that the array should be rotated. If the value is positive the end becomes the start, if it negative it's that start become the end.
+    /// - Returns: self after rotating.
+    @discardableResult
+    public mutating func rotate(by places: Int) -> Self {
+        guard places != 0 && places < count else { return self }
```

Comment:
Isn't it possible to use `places % count` to work with situations of `places >= count`?

Reply:
Yeap, but I'm not sure if this could be a behaviour to implement inside the method :))

Reply:
It just seems like unexpected behaviour to me to `return self` just because the rotation size was too big... It's also troublesome that it doesn't check `abs(places) < count` because this is designed to handle both positive and negative values, and if the negative value is too high this method will crash...
I know you're just moving this method, but while we're here might as well fix it, no? :)

Reply:
For sure it's a nice catch @guykogus the negative too big. I will fix it :)) 
But to me seems weird for thinking in an API usage, for example, have a collection of 4, send 5 but it rotates 1. Maybe if it's well documented, but still I think the expected behavior is still not rotate it :)

Reply:
It doesn't rotate 1, it rotates 5, but since the size is 4 then the final result is equivalent to a rotation of 1.
The idea is to think of the array as a circular container, and when you spin a circle 540 degrees it's the same as spinning it 180, the final result is the same. If I call a method to rotate something 540 degrees and the final result is that it doesn't spin at all because the maximum acceptable input is 360, I would find that _very_ surprising. I think that it's the same here.

Reply:
I see... It makes sense and It seems like this is the same behavior on [ruby api](https://apidock.com/ruby/Array/rotate). So let's change it then :))